hudi-hadoop-common/src/test/java/org/apache/hudi/common/functional/TestHoodieLogFormat.java

```diff
@@ -647,23 +662,185 @@ public void testBasicAppendAndScanMultipleFiles(ExternalSpillableMap.DiskMapType
                                                   boolean enableOptimizedLogBlocksScan)
       throws IOException, URISyntaxException, InterruptedException {
 
-    // Generate 4 delta-log files w/ random records
     Schema schema = HoodieAvroUtils.addMetadataFields(getSimpleSchema());
     SchemaTestUtil testUtil = new SchemaTestUtil();
+    appendAndValidate(schema, testUtil, diskMapType, isCompressionEnabled, enableOptimizedLogBlocksScan,
+        "100");
+  }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void testLogRecordReaderWithMaxInstantTimeConfigured(boolean enableOptimizedLogScan) throws IOException, URISyntaxException, InterruptedException {
+    Schema schema = HoodieAvroUtils.addMetadataFields(getSimpleSchema());
+    SchemaTestUtil testUtil = new SchemaTestUtil();
+
+    Pair<List<IndexedRecord>, Set<HoodieLogFile>> firstBatch = appendAndValidate(schema, testUtil, ExternalSpillableMap.DiskMapType.BITCASK, false, enableOptimizedLogScan,
```

Comment:
since we move to completion time file slicing, it's better to add real timestamp as the instant time instead of integers in the future to avoid corner cases.